aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,125 @@
+
+Client side validation
+----------------------
+Client side validation is a great convenience for users. It saves time they would otherwise spend waiting for a round trip to the server. In business terms, even a few seconds is multiplied hundreds of times each day and adds up to be a lot of time, expense, and frustration. Straightforward and immediate validation enables users to work more efficiently and produce better quality input and output. 
+
+MVC uses data annotations in addition to type metadata from model properties to validate data and display any error messages. It does so by rendering `HTML5 ``data-`` attributes  <http://w3c.github.io/html/dom.html#embedding-custom-non-visible-data-with-the-data-attributes>`_ in the form elements that need validation, as shown below. MVC generates the ``data-`` attributes for both built-in and custom attributes. The ``data-val-required`` attribute below contains an error message to display if the user doesn't fill in the release date field, and that message displays in the accompanying ``<span>`` element.
+
+.. code-block:: html
+ :emphasize-lines: 3
+		
+  <input name="ReleaseDate" class="form-control" id="ReleaseDate" 
+  type="datetime" value="" 
+  data-val-required="The ReleaseDate field is required." 
+  data-val="true">
+        
+  <span class="text-danger field-validation-valid" 
+  data-valmsg-replace="true" 
+  data-valmsg-for="ReleaseDate">
+            
+After the form fields pass client side validation, the ``ModelStateDictionary`` is populated with values. When the HTTP request reaches MVC it invokes the appropriate action method for processing.
+
+Remote valiation
+----------------
+Remote validation is a great feature to use when you need to validate data on the client against data on the server. For example, your app may need to verify whether an email or user name is already in use, and it must query a large amount of data to do so. Downloading large sets of data for validating one field consumes too many resources to be worthwile. An alternative is to make a round-trip request to validate a field; however, doing so diminishes the user's experience, especially when other fields quickly pass client side validation. The solution is to use remote validation that will make the UX better with instant validation and high performance.
```

Comment:
- "worthwhile"? More important that downloading that information may be wasteful and expose sensitive information.
- UX isn't introduced.